cni/pkg/ambient/net_linux.go

```diff
@@ -456,6 +456,19 @@ func (s *Server) CreateRulesOnNode(ztunnelVeth, ztunnelIP string, captureDNS boo
 			"--mark", constants.OutboundMark,
 			"-j", "ACCEPT",
 		),
+
+		// If we have an outbound mark, we don't need kube-proxy to do anything,
```

Comment:
In GCP, I observed that if traffic is being routed to a VM using a k8s svc that doesn't have k8s endpoints objects, it gets dropped by an existing chain with a destination unreachable code. Without this, VM traffic would be dropped.

Reply:
It varies - e.g. with `kindnet` you're correct, I don't see any KUBE-SERVICE drops.
 
 With GCP and other setups we have seen this,  so we added the rule - it's performing the same function as the existing rules above it, which are basically adding the same exclusions for PRERT and POSTRT (FILTER would sit in the middle of those, so skipping it makes as much sense as skipping either end of the same chain).

Reply:
What is "GCP"? There are 3 CNIs available.

Where does it get dropped? Before ztunnel or after?

Reply:
I don't have a live env, but the rule looked like this in the Filter table:
```
-A KUBE-SERVICES -d 192.168.135.127/32 -p tcp -m comment --comment "vmtest/curl:http has no endpoints" -m tcp --dport 9000 -j REJECT --reject-with icmp-port-unreachable
```

Reply:
> Where does it get dropped? Before ztunnel or after?

This is the outbound mark, so it would be after.

Reply:
A request should never go to a service IP _from_ ztunnel

Reply:
`This is the outbound mark, so it would be after.` my bad - yep, this mark is set after the packet leaves the pod and before it gets to ztunnel - so this is to stop `kube-proxy` from dropping it in between.

Again - it's the same rule as the ones above.